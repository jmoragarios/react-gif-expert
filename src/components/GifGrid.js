import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect( () => {
        getGifs();
    }, []);

    const getGifs = async () => {
        
        const url = 'http://api.giphy.com/v1/gifs/search?q=One%20Piece&limit=10&api_key=F5av68ecrjZl4tx7XtSN679Z8HVx91Bl';
        const response = await fetch( url );
        const { data } = await response.json();
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_large.url
            }
        })
        
        console.log(gifs);
        setImages(gifs);
    }

    return (
        <>
            <h3>{ category }</h3>

            {
                images.map( img => (
                    <GifGridItem
                        key={img.id}
                        {...img}    
                    />
                ))
            }
            
        </>
    )
}
