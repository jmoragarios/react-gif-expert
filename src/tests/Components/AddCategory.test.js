import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Test in <AddCategory/>', () => {

    const setCategories = () => {

    }

    test('should show the component successfully', () => {
        
        const wrapper = shallow(<AddCategory setCategories= { setCategories }/>);

        expect( wrapper ).toMatchSnapshot();

    })
    
})
