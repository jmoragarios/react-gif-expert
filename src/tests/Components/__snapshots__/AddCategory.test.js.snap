// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AddCategory/> should show the component successfully 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search Gifs"
    type="text"
    value=""
  />
</form>
`;
